commit ac4c977d16d843f12901595c91773dddb65768a9
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Thu Aug 12 15:37:29 2010 +0200

    mac80211: remove unused don't-encrypt flag
    
    When MFP is disabled, action frames will not
    be encrypted since they are management frames
    and the only management frames that can then
    be encrypted are authentication frames.
    
    Therefore, setting the don't-encrypt flag on
    action frames is unnecessary.
    
    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 9a35d9e7efd7..f9a317766136 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -1555,8 +1555,6 @@ static int ieee80211_action(struct wiphy *wiphy, struct net_device *dev,
 			return -ENOLINK;
 		break;
 	case NL80211_IFTYPE_STATION:
-		if (!(sdata->u.mgd.flags & IEEE80211_STA_MFP_ENABLED))
-			flags |= IEEE80211_TX_INTFL_DONT_ENCRYPT;
 		break;
 	default:
 		return -EOPNOTSUPP;